master/internal/rm/kubernetesrm/jobs.go

```diff
@@ -2068,6 +2150,210 @@ func (j *jobsService) listTCPRoutesInAllNamespaces(
 	return res, nil
 }
 
+func (j *jobsService) verifyNamespaceExists(namespaceName string) error {
+	_, err := j.clientSet.CoreV1().Namespaces().Get(context.Background(), namespaceName,
+		metaV1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("error finding namespace %s: %w", namespaceName, err)
+	}
+
+	j.podInterfaces[namespaceName] = j.clientSet.CoreV1().Pods(namespaceName)
+	j.configMapInterfaces[namespaceName] = j.clientSet.CoreV1().ConfigMaps(namespaceName)
+	j.jobInterfaces[namespaceName] = j.clientSet.BatchV1().Jobs(namespaceName)
+
+	for _, worker := range j.requestQueueWorkers {
+		worker.podInterface = j.podInterfaces
+		worker.configMapInterfaces = j.configMapInterfaces
+		worker.jobInterface = j.jobInterfaces
+	}
+	return nil
+}
+
+func (j *jobsService) createNamespace(namespaceName string) error {
+	_, err := j.clientSet.CoreV1().Namespaces().Create(
+		context.TODO(),
+		&k8sV1.Namespace{
+			TypeMeta: metaV1.TypeMeta{
+				Kind:       "Namespace",
+				APIVersion: "v1",
+			},
+			ObjectMeta: metaV1.ObjectMeta{
+				Name:   namespaceName,
+				Labels: map[string]string{determinedLabel: namespaceName},
+			},
+		},
+		metaV1.CreateOptions{},
+	)
+	if err != nil {
+		if !strings.Contains(err.Error(), "already exists") {
```

Comment:
Comparing string contents of error messages is bad practice; is there a way to detect this case through something like `errors.Is`?